cas-server-core-authentication/src/main/java/org/jasig/cas/authentication/DefaultAuthenticationResultBuilder.java

```diff
@@ -40,6 +40,14 @@ public DefaultAuthenticationResultBuilder(final PrincipalElectionStrategy princi
     }
 
     @Override
+    public Authentication getInitialAuthentication() {
+        if (this.authentications.isEmpty()) {
+            throw new IllegalArgumentException("Authentication chain is empty; no authentications have been collected yet");
```

Comment:
Should we return null (or J8 `Optional<Authentication>`) instead and let the callsite of this method deal with `nil` result?